messagecatalog/spec.md

```diff
@@ -0,0 +1,1042 @@
+  values are `HTML` and `Markdown` as well as `URL` if the documentation
+  attribute contains a URL pointing to external content.
+- Constraints:
+  - OPTIONAL
+
+#### `createdtimeutc` (message definition)
+
+- Type: `Timestamp`
+- Description: Instant when the Message definition was added to the catalog.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+#### `updatedtimeutc` (message definition)
+
+- Type: `Timestamp`
+- Description: Instant when the Message definition was last updated
+- Constraints:
+  - OPTIONAL
+- Assigned by the server.
+
+### 2.2.5 Message definition version
+
+A message definition version is an object that defines the concrete shape of a
+message in said version. The message definition version inherits all attributes
+of the containing message definition object.
+
+#### `version` (message definition version)
+
+- Type: `Integer`
+- Description: The version of the Message definition. This is a simple counter
+  and tracks the version in the scope of this Message definition within the
+  message definition group. The Message definition document content MAY indicate
+  a different versioning scheme.
+- Constraints:
+  - REQUIRED
+  - Assigned by server.
+- Examples:
+  - 1
+  - 2
+
+#### `createdtimeutc` (message definition version)
+
+- Type: `Timestamp`
+- Description: Instant when the Message definition version was added to the
+  catalog.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+## 3. Message definition formats
+
+This section defines metaschemas for several well-known message and event
+formats and/or protocols. These meta-schemas are meant to be used as the body
+content of a message definition version.
+
+A metaschema is a JSON document containing an object with an object-typed
+property for each logical section of the event/message to be described. For
+instance, for CNCF CloudEvents, there is only an "attributes" section. The AMQP
+metaschema reflects the various sections of the AMQP data model.
+
+The metaschemas use the RFC6570 URI template expression language to indicate
+whether and what content is expected for metadata field/attribute/property
+values. An empty string value permits any content. A 'null' value is equivalent
+to the metadata field/attribute/property not being listed.
+
+### 3.1. CNCF CloudEvents
+
+#### 3.1.1. 'attributes' object
+
+The `attributes` object MAY contain any number of uniquely named attribute
+fields, whereby any name MUST conform with CloudEvents naming rules.
+
+Whenever the name of an attribute field matches that of a well-known CloudEvents
+attribute, its value or the constraints for the value expressed with the regular
+expression MUST match the constraint rules of the CloudEvents attribute.
+
+- A definition for the `type` field is REQUIRED and its value MUST follow all
+  constraints of the `type` attribute of CloudEvents. The `id` of a message
+  definition version using this metaschema MUST match the `type` field value.
```

Comment:
How would this work, if event definitions from multiple groups that contain definitions for the same event type are aggregated into a single group?